python/vyos/ifconfig/interface.py

```diff
@@ -1156,12 +1157,15 @@ def set_mirror(self):
             source_if = next(iter(self._config['is_mirror_intf']))
             config = self._config['is_mirror_intf'][source_if].get('mirror', None)
 
-        # Please do not clear the 'set $? = 0 '. It's meant to force a return of 0
-        # Remove existing mirroring rules
-        delete_tc_cmd  = f'tc qdisc del dev {source_if} handle ffff: ingress 2> /dev/null;'
-        delete_tc_cmd += f'tc qdisc del dev {source_if} handle 1: root prio 2> /dev/null;'
-        delete_tc_cmd += 'set $?=0'
-        self._popen(delete_tc_cmd)
+        # Check configuration stored by old perl code before delete T3782
+        c = ConfigTreeQuery()
```

Comment:
Why retrieve a full copy of the configuration when you have `self._config` available  which is populated in `update()`?

`set_mirror()` is then called from `update()` and has the entire configuration dictionary available.

Reply:
Done